core/src/main/scala/io/gearpump/transport/netty/Client.scala

```diff
@@ -200,11 +200,11 @@ object Client {
     }
   }
 
-  class ClientPipelineFactory(name: String, decoder: MessageDecoder, encoder: MessageEncoder) extends ChannelPipelineFactory {
+  class ClientPipelineFactory(name: String, conf: NettyConfig) extends ChannelPipelineFactory {
```

Comment:
With `conf.newTransportSerializer`, each client has its own serializer for ChannelPipeline. Also, I wonder why a client requires a decoder and server an encoder.